tests/test_query.py

```diff
@@ -210,3 +254,26 @@ async def mock_httpx_get(self, **kwargs):
     assert response["errors"] == []
     assert len(response["responses"]) == 2
     assert "Requests to all nodes succeeded (2/2)" in caplog.text
+
+
+def test_query_without_token_succeeds_when_auth_disabled(
+    monkeypatch,
+    test_app,
+    set_valid_test_federation_nodes,
+    mocked_single_matching_dataset_result,
+    disable_auth,
+):
```

Comment:
Nice! We should in the future also add a "fails if I provide silly token with auth enabled" test. But not for now. Could you make an issue for memory?

Reply:
Ah I guess that's mostly taken care of in `test_security` already, yes?

Reply:
indeed!